In [73]:
import pandas as pd
import numpy as np

In [139]:
df_gender = pd.read_csv('gender_sub_titanic.csv')
df_train = pd.read_csv('train_titanic.csv')
df_test = pd.read_csv('test_titanic.csv')


##### Comenzando con el analisis, eliminamos las columnas que no utilizaremos en el proyecto

In [140]:
df_train.drop(['Ticket'], axis=1, inplace=True)
df_test.drop(['Ticket'], axis=1, inplace=True)

##### enriquecemos los datos de [df_gender] porque estaban incompletos

In [141]:
pass_id = df_train[['PassengerId', 'Survived']]
Passenger = pd.concat([pass_id, df_gender])
Passenger

PassengerId  Survived
0              1         0
1              2         1
2              3         1
3              4         1
4              5         0
..           ...       ...
413         1305         0
414         1306         1
415         1307         0
416         1308         0
417         1309         0

[1309 rows x 2 columns]

In [142]:
df_train.drop(['Survived'], axis=1, inplace=True)
df_train.drop(['Cabin'], axis=1, inplace=True)
df_test.drop(['Cabin'], axis=1, inplace=True)

In [143]:
df_train['Embarked'] = df_train['Embarked'].map({'S':'Southampton', 'C':'Cherbourg', 'Q':'Queenstown'})
df_train

PassengerId  Pclass                                               Name  \
0              1       3                            Braund, Mr. Owen Harris   
1              2       1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2              3       3                             Heikkinen, Miss. Laina   
3              4       1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4              5       3                           Allen, Mr. William Henry   
..           ...     ...                                                ...   
886          887       2                              Montvila, Rev. Juozas   
887          888       1                       Graham, Miss. Margaret Edith   
888          889       3           Johnston, Miss. Catherine Helen "Carrie"   
889          890       1                              Behr, Mr. Karl Howell   
890          891       3                                Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch     Fare     Embarked  
0      male  22.0      1      0   7.2500  Southampton  
1    female  38.0      1      0  71.2833    Cherbourg  
2    female  26.0      0      0   7.9250  Southampton  
3    female  35.0      1      0  53.1000  Southampton  
4      male  35.0      0      0   8.0500  Southampton  
..      ...   ...    ...    ...      ...          ...  
886    male  27.0      0      0  13.0000  Southampton  
887  female  19.0      0      0  30.0000  Southampton  
888  female   NaN      1      2  23.4500  Southampton  
889    male  26.0      0      0  30.0000    Cherbourg  
890    male  32.0      0      0   7.7500   Queenstown  

[891 rows x 9 columns]

In [144]:
df_test['Embarked'] = df_test['Embarked'].map({'S':'Southampton', 'C':'Cherbourg', 'Q':'Queenstown'})
df_test

PassengerId  Pclass                                          Name  \
0            892       3                              Kelly, Mr. James   
1            893       3              Wilkes, Mrs. James (Ellen Needs)   
2            894       2                     Myles, Mr. Thomas Francis   
3            895       3                              Wirz, Mr. Albert   
4            896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)   
..           ...     ...                                           ...   
413         1305       3                            Spector, Mr. Woolf   
414         1306       1                  Oliva y Ocana, Dona. Fermina   
415         1307       3                  Saether, Mr. Simon Sivertsen   
416         1308       3                           Ware, Mr. Frederick   
417         1309       3                      Peter, Master. Michael J   

        Sex   Age  SibSp  Parch      Fare     Embarked  
0      male  34.5      0      0    7.8292   Queenstown  
1    female  47.0      1      0    7.0000  Southampton  
2      male  62.0      0      0    9.6875   Queenstown  
3      male  27.0      0      0    8.6625  Southampton  
4    female  22.0      1      1   12.2875  Southampton  
..      ...   ...    ...    ...       ...          ...  
413    male   NaN      0      0    8.0500  Southampton  
414  female  39.0      0      0  108.9000    Cherbourg  
415    male  38.5      0      0    7.2500  Southampton  
416    male   NaN      0      0    8.0500  Southampton  
417    male   NaN      1      1   22.3583    Cherbourg  

[418 rows x 9 columns]

In [145]:
titan = pd.concat([df_train, df_test])
titan

PassengerId  Pclass                                               Name  \
0              1       3                            Braund, Mr. Owen Harris   
1              2       1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2              3       3                             Heikkinen, Miss. Laina   
3              4       1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4              5       3                           Allen, Mr. William Henry   
..           ...     ...                                                ...   
413         1305       3                                 Spector, Mr. Woolf   
414         1306       1                       Oliva y Ocana, Dona. Fermina   
415         1307       3                       Saether, Mr. Simon Sivertsen   
416         1308       3                                Ware, Mr. Frederick   
417         1309       3                           Peter, Master. Michael J   

        Sex   Age  SibSp  Parch      Fare     Embarked  
0      male  22.0      1      0    7.2500  Southampton  
1    female  38.0      1      0   71.2833    Cherbourg  
2    female  26.0      0      0    7.9250  Southampton  
3    female  35.0      1      0   53.1000  Southampton  
4      male  35.0      0      0    8.0500  Southampton  
..      ...   ...    ...    ...       ...          ...  
413    male   NaN      0      0    8.0500  Southampton  
414  female  39.0      0      0  108.9000    Cherbourg  
415    male  38.5      0      0    7.2500  Southampton  
416    male   NaN      0      0    8.0500  Southampton  
417    male   NaN      1      1   22.3583    Cherbourg  

[1309 rows x 9 columns]

##### Reacomodamos para mejor lectura de los datos 

In [146]:
titan = titan.reindex(columns=['PassengerId', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Pclass', 'Fare', 'Embarked'])

In [147]:
titan[['Sex', 'Embarked']] = titan[['Sex', 'Embarked']].astype('category')

In [148]:
titan.convert_dtypes().dtypes

PassengerId       Int64
Name             string
Sex            category
Age             float64
SibSp             Int64
Parch             Int64
Pclass            Int64
Fare            float64
Embarked       category
dtype: object

##### Llenamos los datos nulos

In [149]:
titan.isnull().any()

PassengerId    False
Name           False
Sex            False
Age             True
SibSp          False
Parch          False
Pclass         False
Fare            True
Embarked        True
dtype: bool

In [150]:
titan[['Age']] = titan[['Age']].fillna(titan[['Age']].median())

In [151]:
titan[['Fare']] = titan[['Fare']].fillna(titan[['Fare']].median())

In [152]:
titan[['Embarked']] = titan[['Embarked']].fillna('Southampton')

In [153]:
titan.isnull().any()

PassengerId    False
Name           False
Sex            False
Age            False
SibSp          False
Parch          False
Pclass         False
Fare           False
Embarked       False
dtype: bool

##### Y comenzamos el analisis

###### ¿Cuántas personas iban en el titanic?

In [207]:
titan.count()

PassengerId    1309
Name           1309
Sex            1309
Age            1309
SibSp          1309
Parch          1309
Pclass         1309
Fare           1309
Embarked       1309
dtype: int64

2208 personas de los cuales eran 1309 pasajeros

###### ¿Cuántos hombres y mujeres sobrevivieron?

In [157]:
titan.groupby(['Sex']).count()

PassengerId  Name  Age  SibSp  Parch  Pclass  Fare  Embarked
Sex                                                                 
female          466   466  466    466    466     466   466       466
male            843   843  843    843    843     843   843       843

In [283]:
 t10[t10['Survived'] == 1]['Sex'].value_counts()

female    385
male      109
Name: Sex, dtype: int64

Sobrevivieron 385 mujeres de 466 y 109 hombres de 843

###### ¿Cuál fue el top 10 de edad que más sobrevieron y el top 10 de edad que no lo lograron?

In [202]:
t10 = pd.merge(titan, Passenger, on='PassengerId')
t10

PassengerId                                               Name     Sex  \
0               1                            Braund, Mr. Owen Harris    male   
1               2  Cumings, Mrs. John Bradley (Florence Briggs Th...  female   
2               3                             Heikkinen, Miss. Laina  female   
3               4       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female   
4               5                           Allen, Mr. William Henry    male   
...           ...                                                ...     ...   
1304         1305                                 Spector, Mr. Woolf    male   
1305         1306                       Oliva y Ocana, Dona. Fermina  female   
1306         1307                       Saether, Mr. Simon Sivertsen    male   
1307         1308                                Ware, Mr. Frederick    male   
1308         1309                           Peter, Master. Michael J    male   

       Age  SibSp  Parch  Pclass      Fare     Embarked  Survived  
0     22.0      1      0       3    7.2500  Southampton         0  
1     38.0      1      0       1   71.2833    Cherbourg         1  
2     26.0      0      0       3    7.9250  Southampton         1  
3     35.0      1      0       1   53.1000  Southampton         1  
4     35.0      0      0       3    8.0500  Southampton         0  
...    ...    ...    ...     ...       ...          ...       ...  
1304  28.0      0      0       3    8.0500  Southampton         0  
1305  39.0      0      0       1  108.9000    Cherbourg         1  
1306  38.5      0      0       3    7.2500  Southampton         0  
1307  28.0      0      0       3    8.0500  Southampton         0  
1308  28.0      1      1       3   22.3583    Cherbourg         0  

[1309 rows x 10 columns]

In [265]:
g2 = t10[['Age']].value_counts().head(10)
g2

Age 
28.0    295
24.0     47
22.0     43
21.0     41
30.0     40
18.0     39
25.0     34
36.0     31
29.0     30
26.0     30
dtype: int64

In [266]:
top101 = t10[t10['Survived'] == 1]['Age'].value_counts().head(10)
top101

28.0    86
22.0    21
24.0    20
18.0    16
36.0    16
30.0    16
27.0    15
35.0    14
29.0    13
45.0    12
Name: Age, dtype: int64

In [267]:
top100 = t10[t10['Survived'] == 0]['Age'].value_counts().head(10)
top100

28.0    209
21.0     33
24.0     27
25.0     27
30.0     24
18.0     23
22.0     22
26.0     20
19.0     17
29.0     17
Name: Age, dtype: int64

815 pasajeros No sobrevivieron y 494 Si lo hicieron

###### ¿Cuántos cargos o títulos iban en el barco?

In [268]:
import re

In [269]:
t10['title'] = t10['Name'].apply(lambda x: re.findall("\,\s([\w\s]+)+\.", x)[0])

In [274]:
t10['title'].value_counts()

Mr              757
Miss            260
Mrs             197
Master           61
Dr                8
Rev               8
Col               4
Mlle              2
Major             2
Ms                2
the Countess      1
Jonkheer          1
Capt              1
Lady              1
Mme               1
Dona              1
Sir               1
Don               1
Name: Title, dtype: int64

In [275]:
t10['Title'].value_counts().count()

18

###### ¿Cuánto es la sumatoria del valor de los tickets en USD?

In [279]:
suu = t10[['Fare']].sum()
suu

Fare    43564.9411
dtype: float64

In [282]:
pound = 1
dolar = 1.42
suu * dolar

Fare    61862.216362
dtype: float64

El valor total de los tickets es de 61862.21